src/libsync/capabilities.cpp

```diff
@@ -239,6 +239,27 @@ bool Capabilities::userStatusSupportsEmoji() const
     return userStatusMap.value("supports_emoji", false).toBool();
 }
 
+QColor Capabilities::serverColor() const
+{
+    if (!_capabilities.contains("theming")) {
+        return {};
+    }
+
+    const auto themingMap = _capabilities["theming"].toMap();
+    return QColor(themingMap["color"].toString());
```

Comment:
can you factor the common part in a single utility private method ?
`Capabilities::serverColor()` and `Capabilities::serverTextColor()` share a lot of common code

Reply:
Okay, done